File: fl_amap_map/ios/Classes/AMapEvent.swift
import Flutter
import Foundation

class AMapEvent: NSObject, FlutterStreamHandler {
    var eventSink: FlutterEventSink?
    var eventChannel: FlutterEventChannel?

    init(_ messenger: FlutterBinaryMessenger) {
        super.init()
        eventChannel = FlutterEventChannel(name: "fl_amap/event", binaryMessenger: messenger)
        eventChannel!.setStreamHandler(self)
    }

    func sendEvent(_ arguments: Any?) {
        let mainQueue = DispatchQueue.main
        mainQueue.async {
            self.eventSink?(arguments)
        }
    }

    func dispose() {
        eventSink = nil
        if eventChannel != nil {
            eventChannel?.setStreamHandler(nil)
            eventChannel = nil
        }
    }

    internal func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        return nil
    }

    internal func onCancel(withArguments arguments: Any?) -> FlutterError? {
        dispose()
        return nil
    }
}

File: fl_amap_map/android/src/main/kotlin/fl/amap/AMapEvent.kt
package fl.amap

import android.os.Handler
import android.os.Looper
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.EventChannel

class AMapEvent(binaryMessenger: BinaryMessenger) : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private var eventChannel: EventChannel? = null
    private val handler = Handler(Looper.getMainLooper())

    init {
        eventChannel = EventChannel(binaryMessenger, "fl_amap/event")
        eventChannel!!.setStreamHandler(this)
    }

    fun sendEvent(arguments: Any?) {
        handler.post {
            eventSink?.success(arguments)
        }
    }

    fun dispose() {
        eventSink?.endOfStream()
        eventSink = null
        eventChannel?.setStreamHandler(null)
        eventChannel = null
    }

    override fun onListen(arguments: Any?, event: EventChannel.EventSink?) {
        eventSink = event
    }

    override fun onCancel(arguments: Any?) {
        dispose()
    }
}